article/background.tex

```diff
@@ -4,107 +4,69 @@ \section{Background}
-Containers enable a complete working environment as small as a few Megabytes, and can be started as quickly as a normal process.
-Many container images are publicly available via public image repositories.
-
-Containers technology is not a recent invention, but the term "container" gained popularity alongside the Docker toolset.
-Over time Docker and other organizations have come together under a Linux Foundation project, the "Open Container Initiative" (OCI).
-The OCI governing body has produced an open specification for containers, which can be used by various container runtimes and toolsets.
-OCI complient container images in most cases can be executed identically with Docker, Podman, or other OCI compliant tools.
-
-While OCI images are nearly ubiquitous in the industry, Singularity (recently renamed to Apptainer) is a toolset that was developed specifically for High Performance Computing.
-Singularity has support for converting OCI images into singularity images, and recent versions of Apptainer have also added support to natively run OCI containers.
-Podman apears to be gaining traction in the HPC community, but Apptainer is still required on many systems.
-
-One of the most significant downsides to using Docker in HPC environments was that it required root privilages.
-However, recent advances in container technology have made this unnecessary, and it is now considered best practice to run containers without root privilages when reasonable.
-
-While the original reference OPFVTA article did not leverage this technology, containers can be used improve to improve the reliability and portability of the OPFVTA project.
-In this article, the authors will provide a snapshot of OPFVTA functioning at a certain point in time — mitigating process fragility in view of incrementing software dependency versions.
+Independent verification of published results is a crucial step in the establishment and maintanence of trust in shared scientific understanding. \cite{TODO Too corny to cite Kuhn?}.
+Verification research has become more common due in part to dramatic results \cite{TODO: Salmon}, dedicated communities \cite{TODO: something wholesome}, and widely publicized large-scale reproducibility meta-analyses \cite{RPP}.
+Unfortunately, in Neuroimaging, the execution of data analysis code often remains non-trivial and time consuming. \cite{TODO: I made this up} 
+The execution of data analysis often requires significant knowledge of their software dependencies, which should be considered out-of-scope for the expected requirements of a researcher to reexecute.
+Because the installation, configuration, and execution software environments frequently requires in-depth knowledge of the internal workings, reexecution results remain underrepresented in scientific literature \cite{TODO my opinion, is that ok?}
```

Comment:
There is some grammatical mismatch between the noun and the verb. We don't need to cite sources for everything, what I think you're trying to describe is a logical conclusion of a statement. But I think just referring to environment complexity makes it easier than going into details. It's also not the installation which is tricky, `make install` usually means “copy stuff”. What's tricky is compiling (might need a specific toolchain), and linking (static linking requires a package manager to keep a record of what dependency versions were present at *build* time, Portage does this; dynamic linking means you need to be aware of dependency requirements for future updates lest you break something, Portage does this as well). But really we shouldn't discuss this in any amount of detail...

Reply:
And even though portage does lots of nice things, it still took us a long time to fix our dependencies. 

I am using installation to mean the entire setup of the environment, code, etc. I think its common usage of "install" to include dependency resolution and compilation.